src/reducers/Reports/index.js

```diff
@@ -2,13 +2,20 @@ import * as types from '@constants/Types';
 
 const INITIAL_STATE = {
     fetchingReport: false,
-    reportsList: []
-}
+    reportsList: [],
+    activeReport: null
+};
 
 export default (state = INITIAL_STATE, action) => {
     switch (action.type) {
-        case types.TOGGLE_FETCHING_STATUS: return { ...state, fetchingReport: !state.fetchingReport }
-        case types.SET_CURRENCIES_DATA: return { ...state, reportsList: state.reportsList.concat(action.payload) }
+        case types.TOGGLE_FETCHING_STATUS:
+            return {...state, fetchingReport: !state.fetchingReport};
+        case types.SET_CURRENCIES_DATA:
+            return {...state, reportsList: state.reportsList.concat(action.payload)};
+        case types.SET_ACTIVE_REPORT:
+            return {...state, activeReport: action.payload};
+        case types.DELETE_ACTIVE_REPORT:
+            return {...state, activeReport: null, reportsList: state.reportsList.filter(k => k !== state.activeReport)}

```

Comment:
I often advice to use single action for single job, in order to make clear separation of concerns inside the Redux reducers. It makes app easier to test and also debug.